module/VuFind/tests/unit-tests/src/VuFindTest/Record/LoaderTest.php

```diff
@@ -136,14 +183,24 @@ public function testSingleRecordWithBackendParameters()
 
         $driver = $this->getDriver();
         $collection = $this->getCollection([$driver]);
-        $service = $this->createMock(\VuFindSearch\Service::class);
-        $service->expects($this->once())->method('retrieve')
-            ->with(
-                $this->equalTo('Solr'),
-                $this->equalTo('test'),
-                $this->equalTo($params)
-            )
+
+        $commandObj = $this->getMockBuilder(\VuFindSearch\Command\AbstractBase::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $commandObj->expects($this->once())->method('getResult')
             ->will($this->returnValue($collection));
+        $service = $this->getMockBuilder(\VuFindSearch\Service::class)
+            ->disableOriginalConstructor()->getMock();
+        $checkCommand = function ($command) {
+            return get_class($command) === \VuFindSearch\Command\RetrieveCommand::class
+                && $command->getTargetIdentifier() === "Solr"
+                && $command->getArguments()[0] === "test"
+                && $command->getArguments()[1]->getArrayCopy() === ['fq'=> ['id:test']];
```

Comment:
I also wonder if this test might be simplified (or at least made more readable) by adding a protected support method like:

```
protected function getCommandChecker($class, $args = [], $target = 'Solr')
{
    return function ($command) use ($class, $args, $target) {
        return get_class($command) === $class
            && $command->getArguments() === $args
            && $command->getTargetIdentifer() === $target;
    };
}
```

Then, instead of building custom functions repeatedly, you can just do things like `->with($this->callback($this->getCommandChecker(...)))`.

This is an entirely optional suggestion, and maybe the argument comparison won't work as well as I hope it will... but something you can consider if you like!